src/main/java/jeff/parser/Parser.java

```diff
@@ -0,0 +1,114 @@
+            return new ListCommand(input);
+        } else if (input.equals("bye")) {
+            return new ExitCommand(input);
+        } else if (input.startsWith("mark")) {
+            return new MarkCommand(input);
+        } else if (input.startsWith("unmark")) {
+            return new UnmarkCommand(input);
+        } else if (input.startsWith("delete")) {
+            return new DeleteCommand(input);
+        } else if (input.startsWith("todo") || input.startsWith("deadline") || input.startsWith("event")) {
+            return new AddCommand(input);
+        } else if (input.startsWith("task")) {
+            return new DateCommand(input);
+        } else if (input.startsWith("find")) {
+            return new FindCommand(input);
+        } else {
+            throw new JeffException();
+        }
+    }
+
+    /**
+     * Returns a LocalDateTime object based on the user's input.
+     *
+     * @param input User's input.
+     * @return a LocalDateTime object with the date and time specified by the user.
+     * @throws DateTimeParseException if the user's input is in the wrong format.
+     */
+    public static LocalDateTime getLocalDateTime(String input) throws DateTimeParseException {
+        try {
+            return LocalDateTime.parse(input, DateTimeFormatter.ofPattern("yyyy-MM-dd hh:mm a"));
+        } catch (DateTimeParseException e) {
+            return LocalDateTime.parse(input, DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm"));
+        }
+    }
+
+    /**
+     * Returns the string representation of the given list of tasks.
+     *
+     * @param list Given list of tasks.
+     * @return String representation of the given list.
+     */
+    public static String listToString(List<Task> list) {
+        // Initialise a StringBuilder
+        StringBuilder listString = new StringBuilder();
+
+        // Loop through the inputList and add it to the StringBuilder
+        for (int i = 0; i < list.size(); i++) {
+            listString.append(Integer.toString(i + 1)).append(".").append(list.get(i).toString());
+
+            // Only add a new line when it is not the last task in the list
+            if (i != list.size() - 1) {
+                listString.append("\n");
+            }
+
+        }
+
+        return listString.toString();
+    }
+
+    /**
+     * Adds a space in front of each line in the text.
+     *
+     * @param text Given text.
+     * @return Same text but with a space in front of each line.
+     */
+    public static String prettyText(String text) {
+        // Split the text into lines
+        String[] lines = text.split("\n");
+
+        // Initialise a StringBuilder
+        StringBuilder resultText = new StringBuilder();
+
+        // Add a space in front each line
+        for (String line : lines) {
+            resultText.append(" ").append(line).append("\n");
+        }
+
+        // Convert the StringBuilder back to a String
+        return resultText.toString();
+    }
```

Comment:
The method name `prettyText` might not be the best way to name this static method.

1. Although the word 'pretty' can be used as a verb, it may not be the most intuitive interpretation of the word ('pretty' can be an adjective to describe the word 'text' as well).
2. Moreover, the method name also may not be most descriptive in telling readers what the function does, which is to add a leading space to every line in the string.

Perhaps you can consider renaming the method to
- `convertToPrettyText()` (method name that starts with a verb) or
- `toPrettyText()` (similar to the `toString()` method name)

to address issue 1, or you could consider something even more descriptive (and possible overkill), such as:
- `addLeadingSpaceToText()` (very explicit in terms of what the method does)

to address issue 2.

This ultimately boils down to personal preference, but these would be my suggestions. :)

Reply:
Thanks for pointing it out! I like the suggestion for `addLeadingSpaceToText()` so I would be replacing the function name with this.